File: HydrusSharp.Data/Models/Client/NamedJsonDump.cs
using HydrusSharp.Core.Enums;

namespace HydrusSharp.Data.Models.Client
{
    public class NamedJsonDump
    {
        public int DumpTypeId { get; set; }

        public string DumpName { get; set; }

        public int Version { get; set; }

        public int Timestamp { get; set; }

        public string Dump { get;set; }

        public SerialisableType DumpType => (SerialisableType)DumpTypeId;
    }
}

File: HydrusSharp/Core/Scripts/ViewModels/Page/NamedJsonDump.ts
class NamedJsonDump {

    public type: ko.Observable<number>;
    public name: ko.Observable<string>;
    public version: ko.Observable<number>;
    public timestamp: ko.Observable<number>;
    public dump: ko.Observable<string>;
    public notebook: ko.Observable<GuiSessionContainerPageNotebook>;
    public lastModifiedOn: ko.Computed<string>;

    public constructor(view: INamedJsonDump) {

        this.type = ko.observable(view.DumpType);
        this.name = ko.observable(view.DumpName);
        this.version = ko.observable(view.Version);
        this.timestamp = ko.observable(view.Timestamp);
        this.dump = ko.observable(view.Dump);
        this.notebook = ko.observable(new GuiSessionContainerPageNotebook(JSON.parse(view.Dump)));

        this.lastModifiedOn = ko.computed(() => {

            const date = new Date(this.timestamp());
            return date.toString();
        });
    }
}